File: code/mathematical-algorithms/russian_peasant_multiplication/russian_peasant.rs
fn russian_peasant(mut a: u32, mut b: u32) -> u32 {
    let mut res = 0;

    while b > 0 {
        if (b & 1) != 0 {
            res += a;  
        }

        a = a << 1;
        b = b >> 1;
    }
    res
}

fn main() {
    println!("{}", russian_peasant(18, 24));
    println!("{}", russian_peasant(20, 12));
}

File: code/mathematical-algorithms/russian_peasant_multiplication/russian_peasant.php
<?php 
function russianPeasant($a,$b){
    $res = 0;
	while ($a >= 1){
		if($a%2 !== 0 || $b%2 !== 0){
            $res = $res + $b;
		}
        $a = $a/2;
        $b = $b*2;
    }
    return $res;
}

echo russianPeasant(6,4);